android/src/main/java/com/robinpowered/Frequency/AudioSession.java

```diff
@@ -0,0 +1,49 @@
+package com.robinpowered.RNFrequency;
+
+import android.media.AudioTrack;
+
+import com.facebook.react.bridge.Promise;
+
+/**
+ * Created by Ying Hang Eng on 1/24/18.
+ */
+
+class AudioSession {
+    Promise promise;
+    AudioTrack track;
+    static final String TRACK_STOPPED_PLAYING = "TRACK_STOPPED_PLAYING";
+
+    AudioSession(AudioTrack track, Promise promise) {
+        this.track = track;
+        this.promise = promise;
+    }
+
+    void cancel() {
+        // If track is initialized, stop playback and release resources
+        try {
+            track.stop();
+        } catch (IllegalStateException e) {
+            // no-op
+        }
+
+        track.release();
+        track = null;
+
+        promise.reject(TRACK_STOPPED_PLAYING, "Track stopped playing");
+        promise = null;
+    }
+
+    void complete() {
+        track.release();
+        track = null;
```

Comment:
I don't think we need to be setting things to `null` inside this class. The Module using this class should just set _its_ instance to `null`.

Example:

```java
audioSession.complete()
audioSession = null // this is the cleanup.
```